shells/webextension/src/panel.js

```diff
@@ -77,7 +77,11 @@ var config: Props = {
         window.${globalPathToType}.prototype &&
         window.${globalPathToType}.prototype.isReactComponent
       ) {
-        inspect(window.${globalPathToInst}.render);
+        if (window.${globalPathToInst}.REACT_HOT_LOADER_RENDERED_GENERATION) {
```

Comment:
The `REACT_HOT_LOADER_RENDERED_GENERATION` property is assigned by [React Hot Loader](https://github.com/gaearon/react-hot-loader/blob/acb748b4e410f3b72cd1079f741223f3606f2f23/src/reconciler/proxyAdapter.js#L16). Thanks to this property we can now that we are running RHL.

Reply:
It would not work for SFC, as long they are not in the prototype.
The only way to access "original" component is to use `window.${globalPathToInst}.__reactstandin__getCurrent`([link](https://github.com/gaearon/react-hot-loader/blob/master/src/proxy/createClassProxy.js#L306)), but this is something react-dev-tools should not rely on.